iroh/src/node.rs

```diff
@@ -1165,15 +1182,24 @@ impl<D: BaoStore> RpcHandler<D> {
             }
             match msg.tag {
                 SetTagOption::Named(tag) => {
-                    db.set_tag(tag, Some(haf)).await?;
+                    if let Err(err) = db.set_tag(tag, Some(haf)).await {
+                        if let Err(e) = progress.send(DownloadProgress::Abort(err.into())).await {
+                            error!("error sending download progress: {e}");
+                        }
+                    }
+                    return;
                 }
                 SetTagOption::Auto => {
-                    db.create_tag(haf).await?;
+                    if let Err(err) = db.create_tag(haf).await {
+                        if let Err(e) = progress.send(DownloadProgress::Abort(err.into())).await {
+                            error!("error sending download progress: {e}");
+                        }
+                        return;
+                    }
                 }
             }
             drop(temp_pin);
-            progress.send(DownloadProgress::AllDone).await?;
-            anyhow::Ok(())
+            let _ = progress.send(DownloadProgress::AllDone).await;
         });
```

Comment:
we should also return the errors from the closure and check for errors here, currently this function will exit with `Ok` even though it failed

Reply:
I figured that we weren't `await`ing the closure because the intended use case was to return once the download process was set up and expect users to listen for errors on the progress stream.

I refactored so that this was closer to the other methods that rely on progress streams for communciation.